sqlmesh/core/model/definition.py

```diff
@@ -762,13 +773,21 @@ def metadata_hash(self, audits: t.Dict[str, ModelAudit]) -> str:
 
         for audit_name, audit_args in sorted(self.audits, key=lambda a: a[0]):
             metadata.append(audit_name)
-
+            audit = None
             if audit_name in BUILT_IN_AUDITS:
                 for arg_name, arg_value in audit_args.items():
                     metadata.append(arg_name)
                     metadata.append(gen(arg_value))
+            elif audit_name in self.inline_audits:
+                audit = load_audit(self.inline_audits[audit_name], dialect=self.dialect)
```

Comment:
Can we not call `load_audit` twice for inline audits? E.g. can we parse them only once at validation time? Otherwise, consider caching them or something.